### PR TITLE
Minor cleanups to previous commit.

### PR DESCRIPTION
**Pull request by grp**
- Use (const) pointers instead of references to match existing style.
 - Replace switches on types with casting, reduce switch nesting.

### CODE REVIEW

File: Libraries/plist/Tools/PlistBuddy.cpp [hunk 1]
```diff
@@ -206,6 +172,34 @@ CommandHelp()
...
+        return NULL;
+    }
+
+    std::vector<uint8_t> contents;
+    if (!filesystem->read(&contents, path)) {
+        fprintf(stderr, "error: unable to read %s\n", path.c_str());
+        return NULL;
+    }
+
+    auto deserialize = plist::Format::Any::Deserialize(contents);
```
Comment (grp) [inline Libraries/plist/Tools/PlistBuddy.cpp]: The identified format wasn't being used later, so no need to compute it here: the second argument to `plist::Format::Any::Deserialize` is optional and detects the format automatically by default.

File: Libraries/plist/Tools/PlistBuddy.cpp [hunk 2]
```diff
@@ -467,49 +461,48 @@ Delete(plist::Object *object, std::queue<std::string> *keyPath) {
 }
 
 static bool
-Merge(plist::Object *object, std::string &merge_file, std::queue<std::string> &keyPath, Filesystem *filesystem)
+Merge(plist::Object *object, std::string &mergeSource, std::queue<std::string> *keyPath, Filesystem const *filesystem)
```
Comment (Ktwu) [inline Libraries/plist/Tools/PlistBuddy.cpp]: Why make keyPath a pointer?  It's not intended to be mutated.

File: Libraries/plist/Tools/PlistBuddy.cpp (hunk 1 above)
Comment (Ktwu) [inline Libraries/plist/Tools/PlistBuddy.cpp]: Nice, thanks.

File: Libraries/plist/Tools/PlistBuddy.cpp (hunk 2 above)
Comment (grp) [inline Libraries/plist/Tools/PlistBuddy.cpp]: `GetObjectAtKeyPath()` mutates it below, IIRC.

**Pull request merged**